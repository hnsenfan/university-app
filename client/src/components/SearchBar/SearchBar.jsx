import { Fragment } from 'react'
import PropTypes from 'prop-types'

import MiniSearch from '../Icons/General/MiniSearch'

import './SearchBar.scss'

const SearchBar = ({ optionArray, handleChangeValue, handleSubmitSearch }) => {
  return (
    <Fragment>
      <div className='d-flex flex-wrap-wrap justify-content-center'>
        { optionArray.map((option, index) =>
          <div key={index} className='d-flex flex-grow-1 mt-m mr-s'>
            <MiniSearch />
            <input type='search' id={option} placeholder={`Search by ${option}`} onChange={ handleChangeValue }/>
          </div>
        )}
      </div>
      <div className='d-flex justify-content-center mt-l'>
        <button className='btn-primary text-bold text-18' onClick={ handleSubmitSearch }>Search</button>
      </div>
    </Fragment>
  )
}

SearchBar.propTypes = {
  optionArray: PropTypes.array,
  handleChangeValue: PropTypes.func,
  handleSubmitSearch: PropTypes.func
}

export default SearchBar