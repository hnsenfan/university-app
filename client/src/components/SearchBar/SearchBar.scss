@import '/src/modules/styles/core/variables';

.search-page {
  flex-direction: row;
  @media (max-width: $screen-md) {
    flex-direction: column;
  }
}

.search-container {
  flex-basis: 50%;
  @media (max-width: $screen-md) {
    flex-basis: 100%;
  }
}

input[type=search] {
  padding: 12px;
  font-size: 16px;
  text-indent: 24px;
  width: 100%;
}

.radio-background {
  padding: 10px;
  margin-right: 8px;
  background: map-get($colors, white);
  color: map-get($colors, text-dark-grey);
  width: fit-content;
  border-radius: 24px;
  cursor: pointer;
}

.radio-background-selected {
  padding: 10px;
  margin-right: 8px;
  background: map-get($colors, primary);
  color: map-get($colors, white);
  width: fit-content;
  border-radius: 24px;
  cursor: pointer;
}

input[type=radio] {
  display: none;
}