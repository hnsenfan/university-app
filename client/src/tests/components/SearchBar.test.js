import { shallow } from 'enzyme'
import SearchBar from '../../components/SearchBar/SearchBar'

test('should test SearchBar component', () => {
 const wrapper = shallow(<SearchBar />)
 expect(wrapper).toMatchSnapshot()
})

test('should handle handleSubmitSearch handler', () => {
  const onSubmitMock = jest.fn()
  const wrapper = shallow(<SearchBar handleSubmitSearch={onSubmitMock} />)
  expect(wrapper).toMatchSnapshot()
  wrapper.find('.btn-primary').simulate('click')
  expect(onSubmitMock).toHaveBeenCalled()
 })

 test('should handle handleChangeValue handler', () => {
  const onChangeMock = jest.fn()
  const wrapper = shallow(<SearchBar optionArray={['name', 'country']} handleChangeValue={onChangeMock} />)
  wrapper.find('.name').simulate('change', { preventDefault() {}, target: { value: 'new value' } })
  expect(onChangeMock).toHaveBeenCalled()
 })