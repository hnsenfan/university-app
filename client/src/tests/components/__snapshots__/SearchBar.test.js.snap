// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle handleSubmitSearch handler 1`] = `
<Fragment>
  <div
    className="d-flex flex-wrap-wrap justify-content-center"
  />
  <div
    className="d-flex justify-content-center mt-l"
  >
    <button
      className="btn-primary text-bold text-18"
      onClick={[MockFunction]}
    >
      Search
    </button>
  </div>
</Fragment>
`;

exports[`should test SearchBar component 1`] = `
<Fragment>
  <div
    className="d-flex flex-wrap-wrap justify-content-center"
  />
  <div
    className="d-flex justify-content-center mt-l"
  >
    <button
      className="btn-primary text-bold text-18"
    >
      Search
    </button>
  </div>
</Fragment>
`;
